retriever/lib/engine_tools.py

```diff
@@ -97,12 +97,9 @@ def final_cleanup(engine):
 config_path = os.path.join(HOME_DIR, 'connections.config')
 
 
-def reset_retriever(scope="all", ask_permission=True):
+def reset_retriever(data, script, ask_permission=True):
```

Comment:
This is a backwards compatibility breaking change. Alternatively we could use an API like:

`def reset_retriever(scope="all", ask_permission=True, dataset=None):`

and then to reset a dataset call this function as:

`reset_retriever("dataset", dataset = "mammal-masses")`

I doubt `ask_permission` is in active use based on position, so we could also the more natural:

`def reset_retriever(scope="all", dataset=None, ask_permission=True,):`

Allowing:

`reset_retriever("dataset", "mammal-masses")`

Reply:
sorry, @ethanwhite I was a bit occupied, so could not reply. So I would do it in the manner you stated.